main.tf

```diff
@@ -48,3 +61,15 @@ resource "azurerm_log_analytics_data_export_rule" "rule" {
   table_names             = each.value.table_names
   enabled                 = try(each.value.enabled, true)
 }
+
+resource "azurerm_user_assigned_identity" "identity" {
+  for_each = lookup(var.law, "identity", null) != null ? (
+
+    lookup(var.law.identity, "type", null) == "UserAssigned" &&
+    lookup(var.law.identity, "identity_ids", null) == null ? { "identity" = var.law.identity } : {}
```

Comment:
can the "identity" key be replaced with the actual name of the UAI? 
so probably: 
`lookup(var.law, "name", "uai-${var.law.name}")`

Reply:
not currently possible due to the name of the UAI being derived from the LA workspace name, which with name_unique is randomised and only known after apply.

Reply:
yeah it requires a globally unique name